#VRML_SIM R2018a utf8
#Author(s):         Inbae Jeong, Chansol Hong
#Maintainer:        Chansol Hong (cshong@rit.kaist.ac.kr)

WorldInfo {
  ERP 0.15
  physics "collision_detector"
  basicTimeStep 10
  FPS 20
  optimalThreadCount 8
  defaultDamping Damping {
  }
  lineScale 0.01
  contactProperties [
    DEF floor-robot_body ContactProperties {
      material1 "floor"
      material2 "body"
      coulombFriction [
        0
      ]
      bounce 0
      softERP 0.15
      softCFM 1e-5
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF floor-wheel ContactProperties {
      material1 "floor"
      material2 "wheel"
      bounce 0
      softERP 0.15
      softCFM 1e-5
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF floor-ball ContactProperties {
      material1 "floor"
      material2 "ball"
      coulombFriction [
        0.1
      ]
      bounce 0
      softERP 0.15
      softCFM 1e-5
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wall-body ContactProperties {
      material1 "wall"
      material2 "body"
      bounce 0
      softERP 0
      softCFM 1e-5
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wall-wheel ContactProperties {
      material1 "wall"
      material2 "wheel"
      bounce 0
      softERP 0
      softCFM 1e-5
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wall-ball ContactProperties {
      material1 "wall"
      material2 "ball"
      bounce 0.5
      softERP 0.8
      softCFM 1e-9
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF body-body ContactProperties {
      material1 "body"
      material2 "body"
      bounce 0.25
      softERP 0
      softCFM 1e-7
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF body-wheel ContactProperties {
      material1 "body"
      material2 "wheel"
      bounce 0
      softERP 0.3
      softCFM 1e-7
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF body-ball ContactProperties {
      material1 "body"
      material2 "ball"
      bounce 0.5
      softERP 0.15
      softCFM 1e-5
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wheel-wheel ContactProperties {
      material1 "wheel"
      material2 "wheel"
      bounce 0
      softERP 0
      softCFM 1e-5
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wheel-ball ContactProperties {
      material1 "wheel"
      material2 "ball"
      bounce 0
      softERP 0
      softCFM 1e-5
      bumpSound ""
      rollSound ""
      slideSound ""
    }
  ]
}
DEF DEF_AUDVIEW Viewpoint {
  orientation -1 0 0 0.8910057735495215
  position 0 6.839949869285961 5.658156540652288
}
Background {
  skyColor [
    0.7 0.8 1
  ]
}
DirectionalLight {
  ambientIntensity 1
  direction 0 -1 0
}
Referee {
  translation 0 12.72 0
  children [
    DEF DEF_RECV Receiver {
      name "recv"
    }
    Transform {
      rotation 1 0 0 4.712389
      children [
        DEF DEF_CAMA Camera {
          name "cam_a"
          fieldOfView 0.325
          width 640
          height 480
        }
        DEF DEF_CAMB Camera {
          rotation 0 0 1 3.1415926
          name "cam_b"
          fieldOfView 0.325
          width 640
          height 480
        }
      ]
    }
  ]
}
SoccerField5v5 {
  height 0.2
  grassSlot DEF DEF_GRASS SoccerFieldGrass {
  }
  floorContactMaterial "floor"
  wallContactMaterial "wall"
}
DEF DEF_STADIUM Stadium {
  scale 1.65 1.65 1.65
  castShadows FALSE
  bannerUrl [
    "textures/cyberbotics.png"
    "textures/hynix.png"
    "textures/kaist.png"
  ]
  goalPost [
    DEF DEF_GOALPOST Group {
      children [
        Goalpost {
          translation -1.24 0 0
          scale 0.88 0.88 0.88
          castShadows IS castShadows
          name "goalpost_left"
        }
        Goalpost {
          translation 1.24 0 0
          rotation 0 1 0 3.141592635
          scale 0.88 0.88 0.88
          castShadows IS castShadows
          name "goalpost_right"
        }
      ]
    }
  ]
}
DEF DEF_BALL SoccerBall {
  translation 0 0.03 0
  contactMaterial "ball"
  mass 0.0184
  shapes [
    DEF DEF_BALLSHAPE SoccerBallShape {
      radius 0.03
    }
    DEF DEF_ORANGESHAPE SoccerBallOrangeShape {
      radius 0.03
    }
  ]
}
DEF DEF_ROBOTR0 SoccerRobot {
  translation -0.4 0.0375 0.3
  rotation 0 1 0 -1.5708
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "R0"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
}
DEF DEF_ROBOTR1 SoccerRobot {
  translation -0.4 0.0375 -0.3
  rotation 0 1 0 -1.5708
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "R1"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      id 1
    }
    SoccerRobotIDPatch {
      id 31
      name "id_patch_red"
    }
    SoccerRobotIDPatch {
      id 31
      isTeamTagRed FALSE
      name "id_patch_blue"
    }
  ]
}
DEF DEF_ROBOTR2 SoccerRobot {
  translation -1 0.0375 0.6
  rotation 0 1 0 -1.5708
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "R2"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      id 2
    }
    SoccerRobotIDPatch {
      id 227
      name "id_patch_red"
    }
    SoccerRobotIDPatch {
      id 227
      isTeamTagRed FALSE
      name "id_patch_blue"
    }
  ]
}
DEF DEF_ROBOTR3 SoccerRobot {
  translation -1 0.0375 -0.6
  rotation 0 1 0 -1.5708
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "R3"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      id 3
    }
    SoccerRobotIDPatch {
      id 364
      name "id_patch_red"
    }
    SoccerRobotIDPatch {
      id 364
      isTeamTagRed FALSE
      name "id_patch_blue"
    }
  ]
}
DEF DEF_ROBOTR4 SoccerRobot {
  translation -1.8 0.0375 0
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "R4"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      id 4
    }
    SoccerRobotIDPatch {
      id 437
      name "id_patch_red"
    }
    SoccerRobotIDPatch {
      id 437
      isTeamTagRed FALSE
      name "id_patch_blue"
    }
  ]
}
DEF DEF_ROBOTB0 SoccerRobot {
  translation 0.4 0.0375 -0.3
  rotation 0 1 0 1.5708
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "B0"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      isTeamTagRed FALSE
    }
    SoccerRobotIDPatch {
      isTeamTagRed FALSE
      name "id_patch_red"
    }
    SoccerRobotIDPatch {
      name "id_patch_blue"
    }
  ]
}
DEF DEF_ROBOTB1 SoccerRobot {
  translation 0.4 0.0375 0.3
  rotation 0 1 0 1.5708
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "B1"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      id 1
      isTeamTagRed FALSE
    }
    SoccerRobotIDPatch {
      id 31
      isTeamTagRed FALSE
      name "id_patch_red"
    }
    SoccerRobotIDPatch {
      id 31
      name "id_patch_blue"
    }
  ]
}
DEF DEF_ROBOTB2 SoccerRobot {
  translation 1 0.0375 -0.6
  rotation 0 1 0 1.5708
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "B2"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      id 2
      isTeamTagRed FALSE
    }
    SoccerRobotIDPatch {
      id 227
      isTeamTagRed FALSE
      name "id_patch_red"
    }
    SoccerRobotIDPatch {
      id 227
      name "id_patch_blue"
    }
  ]
}
DEF DEF_ROBOTB3 SoccerRobot {
  translation 1 0.0375 0.6
  rotation 0 1 0 1.5708
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "B3"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      id 3
      isTeamTagRed FALSE
    }
    SoccerRobotIDPatch {
      id 364
      isTeamTagRed FALSE
      name "id_patch_red"
    }
    SoccerRobotIDPatch {
      id 364
      name "id_patch_blue"
    }
  ]
}
DEF DEF_ROBOTB4 SoccerRobot {
  translation 1.8 0.0375 0
  rotation 0 1 0 3.14159
  lwRotation 1 0 0 1.5708
  rwRotation 1 0 0 1.5708
  name "B4"
  customData "0 0"
  maxSpeed 50
  slipNoise 0.05
  maxForce 0.1
  bodyContactMaterial "body"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      id 4
      isTeamTagRed FALSE
    }
    SoccerRobotIDPatch {
      id 437
      isTeamTagRed FALSE
      name "id_patch_red"
    }
    SoccerRobotIDPatch {
      id 437
      name "id_patch_blue"
    }
  ]
}
